### ISSUE TITLE
E Checkpoint directory (/path_system/.local/share/deepspeech/ldc93s1) does not contain a valid checkpoint state

### ISSUE THREAD

**Issue by darius1tanz** [mozilla/DeepSpeech#2014]
system Info- Intel© Core™ i7-8700K CPU @ 3.70GHz 
GPU:- 2070 RTX  8GB
RAM:- 32 GiB
OS:- Linux Mint 19.1 Cinnamon
tensorflow gpu version == 1.13.1

Everything working fine but got Following Error also i check in this (/system_path/.local/share/deepspeech/ldc93s1) directory there is no previous checkpoints i don't know why this error occur please help me:-

WARNING:tensorflow:From /home/gpu1/tmp/deepspeech-venv/lib/python3.6/site-packages/tensorflow/contrib/rnn/python/ops/lstm_ops.py:696: to_int64 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
Instructions for updating:
I Initializing variables...
I STARTING Optimization
I Training epoch 0...
2019-04-06 04:11:00.963684: E tensorflow/stream_executor/cuda/cuda_dnn.cc:334] Could not create cudnn handle: CUDNN_STATUS_INTERNAL_ERROR
2019-04-06 04:11:00.968836: E tensorflow/stream_executor/cuda/cuda_dnn.cc:334] Could not create cudnn handle: CUDNN_STATUS_INTERNAL_ERROR

**Comment by tuttlebr** [comment]
I have a similar issue. Trying to train without any checkpoints with a script that worked previously. The error is a log out from working the evaluated. Py file. The call expects checkpoint DIR but the saver saves nothing so exit code. I tried with the 0.4.1 checkpoints just to see if that worked but got the same result.

**Comment by tuttlebr** [comment]
FYI the cudnn error is unrelated to what I'm seeing. Do you have the correct nvidia driver + cuda + cudnn combo for tensorflow 1.13.1?

**Comment by lissyx** [comment]
> system Info- Intel© Core™ i7-8700K CPU @ 3.70GHz
> GPU:- 2070 RTX 8GB
> RAM:- 32 GiB
> OS:- Linux Mint 19.1 Cinnamon
> tensorflow gpu version == 1.13.1
> 
> Everything working fine but got Following Error also i check in this (/system_path/.local/share/deepspeech/ldc93s1) directory there is no previous checkpoints i don't know why this error occur please help me:-
> 
> WARNING:tensorflow:From /home/gpu1/tmp/deepspeech-venv/lib/python3.6/site-packages/tensorflow/contrib/rnn/python/ops/lstm_ops.py:696: to_int64 (from tensorflow.python.ops.math_ops) is deprecated and will be removed in a future version.
> Instructions for updating:
> I Initializing variables...
> I STARTING Optimization
> I Training epoch 0...
> 2019-04-06 04:11:00.963684: E tensorflow/stream_executor/cuda/cuda_dnn.cc:334] Could not create cudnn handle: CUDNN_STATUS_INTERNAL_ERROR
> 2019-04-06 04:11:00.968836: E tensorflow/stream_executor/cuda/cuda_dnn.cc:334] Could not create cudnn handle: CUDNN_STATUS_INTERNAL_ERROR

I see several errors here, can you be more specific ? I can't understand your current state. So far, if your training fails with `CUDNN_STATUS_INTERNAL_ERROR` it means your setup is wrong and it's expected there's no checkpoint created ...

**Comment by tuttlebr** [comment]
I'm having the error stated in the OPs title, not a cuDNN error.

Input: running a shell script with my hyperparameters
```bash
#!/bin/bash

set -xe
if [ ! -f DeepSpeech.py ]; then
    echo "Assert: Hi. Please run this from deepSpeech top level directory."
    exit 1
fi;

python3 -u DeepSpeech.py \
  --train_files '/audiodata/bucket1/AudioData/Utterances/valid-train.csv' \
  --dev_files '/audiodata/bucket1/AudioData/Utterances/valid-dev.csv' \
  --test_files '/audiodata/bucket1/AudioData/Utterances/valid-test.csv' \
  --train_batch_size 64 \
  --dev_batch_size 48 \
  --test_batch_size 48 \
  --epochs 75 \
  --dropout_rate .1 \
  --learning_rate 0.0001 \
  --export_dir '/audiodata/bucket1/AudioData/Utterances/SpeechModel/' \
  --alphabet_config_path '/audiodata/bucket1/AudioData/Utterances/SpeechModel/alphabet.txt' \
  --lm_binary_path '/audiodata/bucket1/AudioData/Utterances/SpeechModel/lm.binary' \
  --lm_trie_path '/audiodata/bucket1/AudioData/Utterances/SpeechModel/trie' \
  --report_count 10 \
  "$@"```


output from console:
```
+ '[' '!' -f DeepSpeech.py ']'
+ python3 -u DeepSpeech.py --train_files /audiodata/bucket1/AudioData/Utterances/valid-train.csv --dev_files /audiodata/bucket1/AudioData/Utterances/valid-dev.csv --test_files /audiodata/bucket1/AudioData/Utterances/valid-test.csv --train_batch_size 64 --dev_batch_size 48 --test_batch_size 48 --epochs 25 --dropout_rate .1 --learning_rate 0.00001 --export_dir /audiodata/bucket1/AudioData/Utterances/SpeechModel/ --checkpoint_dir /audiodata/bucket1/AudioData/Utterances/SpeechModel/checkpoints/ --alphabet_config_path /audiodata/bucket1/AudioData/Utterances/SpeechModel/alphabet.txt --lm_binary_path /audiodata/bucket1/AudioData/Utterances/SpeechModel/lm.binary --lm_trie_path /audiodata/bucket1/AudioData/Utterances/SpeechModel/trie --report_count 10
I Initializing variables...
I STARTING Optimization
I Training epoch 0...
E Checkpoint directory (/audiodata/bucket1/AudioData/Utterances/SpeechModel/checkpoints/) does not contain a valid checkpoint state.
  0% (1 of 1604) |                       | Elapsed Time: 0:00:05 ETA:   2:18:24
```

In reading the flags.py file in more detail, there is a [new flag](https://github.com/mozilla/DeepSpeech/blob/15adf008ae0cf93d0a167a796eb541fb1b43022a/util/flags.py#L70).  The default is auto, which attempts to figure out what I want given the contents in the directory provided. If I set this to 'init' the exit code persists, training stops.

I've also tried various combinations/omissions of checkpoint_dir, and fine_tuning_checkpints, all have same issue.

**Comment by lissyx** [comment]
@tuttlebr You don't specify any checkpoint directory, so the default one applies and should be `$HOME/.local/share/deepspeech/ldc93s1/`

**Comment by reuben** [comment]
If you're trying to fine tune our release checkpoints, they're no longer compatible with master code due to a recent refactoring to remove distributed training. Try specifying an empty checkpoint_dir (and training from scratch), or downgrade the code to an older version.

**Comment by reuben** [comment]
The `tf.train.Coordinator.stop_on_exception()` context manager is swallowing the underlying problem here, try removing it to see what is actually causing things to fail (fixed in #2023).

**Comment by tuttlebr** [comment]
Ah-ha! It ended up being the output > input error thrown by the ctc loss function. For now, i'll just change to ignore_longer_outputs_than_inputs=True. 


`InvalidArgumentError (see above for traceback): Not enough time for target transition sequence (required: 5, available: 4)3You can turn this error into a warning by using the flag ignore_longer_outputs_than_inputs
	 [[node tower_0/CTCLoss (defined at DeepSpeech.py:188)  = CTCLoss[ctc_merge_repeated=true, ignore_longer_outputs_than_inputs=false, preprocess_collapse_repeated=false, _device="/job:localhost/replica:0/task:0/device:CPU:0"](tower_0/raw_logits/_89, tower_0/DeserializeSparse, tower_0/DeserializeSparse:1, tower_0/IteratorGetNext:1)]]
	 [[{{node Mean_8/_145}} = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device_incarnation=1, tensor_name="edge_714_Mean_8", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:GPU:0"]()]]`


I have a check for this if others experience it, but I'm pretty sure the gradient is set to 0 if this error occurs and is ignored. It uses components from your old preprocessing.py file. 


```def is_this_bad_audio(audio_filename, transcript):
    r"""
    A simple audio validation process to determine if an audio sample may be a good candidate 
    for CTC loss given audio represented as MFCC and transcript represented as vector. Poor
    audio will also be removed by determining the decibel level averages across the audio file.
    """
    try:
        features_len = len(audiofile_to_input_vector(audio_filename, numcep=26, numcontext=9)) - 2 * 9 #features_len
        transcript_len = len(text_to_char_array(transcript, alphabet_))
        if features_len < transcript_len:
            answer = "CTCError"
        elif features_len <= 1:
            answer = "TransLen<=1"
        else:
            q = pydub.AudioSegment.from_wav(audio_filename)
            if str(q.dBFS) == '-inf':
                answer = 'dBFS-inf'
            elif q.split_to_mono()[0].max <= 500:
                answer = "LowAmplitude"
            else:
                answer = "OK"
    except:
        answer = "FileError"
            
    return answer```

**Comment by 20richardh** [comment]
> If you're trying to fine tune our release checkpoints, they're no longer compatible with master code due to a recent refactoring to remove distributed training. Try specifying an empty checkpoint_dir (and training from scratch), or downgrade the code to an older version.

Does that mean that the 0.4.1 checkpoint no longer works during training?

**Comment by lissyx** [comment]
> So, this means that the 0.4.1 checkpoint no longer works if you want to use it for additional training? About when will there be a compatible version if so?

That's done now.
